hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -205,6 +218,9 @@ class IncrementalRelation(val sqlContext: SQLContext,
       val endInstantArchived = commitTimeline.isBeforeTimelineStarts(endInstantTime)
 
       val scanDf = if (fallbackToFullTableScan && (startInstantArchived || endInstantArchived)) {
+        if (useStateTransitionTime) {
+          throw new HoodieException("Cannot use stateTransitionTime while enables full table scan")
```

Comment:
throw error or could we log a warn and move on?

Reply:
If enables `useStateTransitionTime`, means they provides state transition time for `startInstantTime` and `endInstantTime`, and they will be used to compare with `_hoodie_commit_time`, then the result is not accurate. So here we better then error out, or force setting `useStateTransitionTime` back to false?

Reply:
I am not sure if we can throw and move on. 
I see this similar to how someone is doing an incremental query as of now. 
what we are handling here is: 
if downstream is lagging /down for sometime and then it resumes to consume, and if cleaner has cleaned up the instants to consume, we do a fallback here. 

when migrating to stateTransitionTime, we should find a way to support the same.